Rework of keywords from d3 10 Advanced pick

glenkleidon opened the pull request:
I did a fair amount of work to support new keywords and structures in D3 10.x Advance pick basic about 4 years ago.  I forgot I hadn't done a pull request - just revisited today.
  
There are a number of revisions which loose some of your specific changes (eg GOTO), but this is implemented in a more general way with gosub and other branching features.

glenkleidon committed:
Rework of keywors from d3 10 Advanced pick

glenkleidon commented:
Hmm, Actually I see there are some keywords from U2/Universe such as CallDotNet and CallJ that I have dropped.  I will look at this again later today.

glenkleidon committed:
INTERIM CHECKIN Re-adding JBASE commands
Still some work to do.

glenkleidon committed:
Completed Re-integration of JBASE and U2/Universe commands.
Fixed JBase Threads, UPCASE XTD

mallek commented:
Thank you glen for your work on this.  I am a little overwhelmed at work trying to catch up.  I will do my best to review and test this over the weekend

glenkleidon committed:
Fixed IF THEN ELSE - moved to block section.

glenkleidon commented:
OK. Reintegration of Jbase and U2/Universe commands.  Fixed some minor issues eg JBASEThreads and XTD & UPCASE.

I think this is complete now.

glenkleidon committed:
Added Rem to comment config comment.
There does not appear to be any significant different with or without the setting however.
